acme.go

```diff
@@ -116,7 +117,8 @@ type AuthzID struct {
 // Client implements ACME spec.
 type Client struct {
 	http.Client
-	Key *rsa.PrivateKey
+	// Key.Public() must be *rsa.PublicKey
+	Key crypto.Signer
```

Comment:
I'm open to returning error values for invalid key types, though, but I wanted to find out whether anyone cared about PRs here before I started threading errors through everything.